### PR TITLE
Enable to stop in struct's IterFields method

### PR DESCRIPTION
**Pull request by sairoutine**
As well as list and map, I would like to write the following the code by struct's iteration method.
```
var error err
str.IterFields(func(name string, value types.Value) bool {
    err = SomeMethod()
    if err != nil {
        return true
    }
    return false
})

if err != nil {
    return err
}
```

### CODE REVIEW

File: go/types/struct_test.go [hunk 1]
```diff
@@ -330,3 +330,39 @@ func TestStructWithNil(t *testing.T) {
...
+	i := 0
+	tstruct.IterFields(func(k string, v Value) bool {
+		assert.True(tstruct.Get(k).Equals(v))
+
+		i += 1
+
+		return false
+	})
+
+	assert.True(i == 3)
```
Comment (aboodman) [inline go/types/struct_test.go]:
```suggestion
	assert.Equal(3, i)
```

File: go/types/struct_test.go [hunk 2]
```diff
@@ -330,3 +330,39 @@ func TestStructWithNil(t *testing.T) {
...
+		if k == "b" {
+			return true
+		}
+
+		i += 1
+
+		return false
+	})
+
+	assert.True(i == 1)
```
Comment (aboodman) [inline go/types/struct_test.go]:
```suggestion
	assert.Equal(1, i)
```

**Comment by sairoutine** [comment]
I finished the agreement.

File: go/types/struct_test.go (hunk 1 above)
Comment (sairoutine) [reply go/types/struct_test.go]: Fixed

File: go/types/struct_test.go (hunk 2 above)
Comment (sairoutine) [reply go/types/struct_test.go]: Fixed

**Comment by sairoutine** [comment]
It seems that it is not the matter of this pull request that fails the test of Travis.

**Comment by sairoutine** [comment]
@aboodman 
Thank you for your review. I finished fixing your pointed out.
Please review again.

**Comment by arv** [comment]
Wasn't there some performance issues with assert.Equal?

**Comment by aboodman** [comment]
@arv is right - I forgot about that perf issue. I will land the version of the change that you originally posted.

Thanks for the patch!

**Comment by aboodman** [comment]
Actually the usage here is in a test file, so the perf is irrelevant. Landed as-is.

**Comment by sairoutine** [comment]
Thank you for your merging!

**Pull request merged**